cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_brute_force_dpc.cpp

```diff
@@ -43,27 +43,42 @@ using dal::backend::context_gpu;
 template <typename Task>
 using descriptor_t = detail::descriptor_base<Task>;
 
+using voting_t = ::oneapi::dal::knn::voting_mode;
+
 namespace de = ::oneapi::dal::detail;
 namespace bk = ::oneapi::dal::backend;
 namespace pr = ::oneapi::dal::backend::primitives;
 
-template <typename Float>
-sycl::event sqrt(sycl::queue& q, array<Float>& data, const bk::event_vector& deps = {}) {
-    ONEDAL_ASSERT(data.has_mutable_data());
-    const auto length = data.get_count();
-    const auto range = bk::make_range_1d(length);
-    auto* const data_ptr = data.get_mutable_data();
+template <typename T1, typename T2>
+sycl::event copy_with_sqrt(sycl::queue& q,
+                           pr::ndview<T1, 2>& dst,
+                           const pr::ndview<T2, 2>& src,
+                           const bk::event_vector& deps = {}) {
+    ONEDAL_ASSERT(src.has_data());
```

Comment:
assert that this function works only with fptypes?

Reply:
1) Even if it works both for integers and fptypes (wider range of parameters) I see no reason to change this. 
2) It should throw compilation error due to the standard: `sycl::sqrt`